zetta_utils/layer/volumetric/cloudvol/backend.py

```diff
@@ -213,3 +235,71 @@ def clone(self, **kwargs) -> CVBackend:
             if k not in implemented_keys:
                 raise KeyError(f"key {k} received, expected one of {implemented_keys}")
         return attrs.evolve(deepcopy(self), path=kwargs["name"])
+
+    def get_voxel_offset(self, resolution: Vec3D) -> IntVec3D:
+        cvol = get_cv_cached(cloudpath=self.path, mip=tuple(resolution), **self.cv_kwargs)
+        return IntVec3D(*cvol.voxel_offset)
+
+    def set_voxel_offset(self, voxel_offset: IntVec3D, resolution: Vec3D) -> None:
+        assert self.info_spec is not None
+        key = "_".join([_str(v) for v in resolution])
+        if self.info_spec.voxel_offset_map is None:
+            self.info_spec.voxel_offset_map = {}
+        self.info_spec.voxel_offset_map[key] = voxel_offset
+        self.on_info_exists = "overwrite"
+        self.__attrs_post_init__()
+
+    def get_chunk_size(self, resolution: Vec3D) -> IntVec3D:
+        cvol = get_cv_cached(cloudpath=self.path, mip=tuple(resolution), **self.cv_kwargs)
+        return IntVec3D(*cvol.chunk_size)
+
+    def set_chunk_size(self, chunk_size: IntVec3D, resolution: Vec3D) -> None:
+        assert self.info_spec is not None
+        key = "_".join([_str(v) for v in resolution])
+        if self.info_spec.chunk_size_map is None:
+            self.info_spec.chunk_size_map = {}
+        self.info_spec.chunk_size_map[key] = chunk_size
+        self.on_info_exists = "overwrite"
+        self.__attrs_post_init__()
+
+    def get_chunk_aligned_index(  # pragma: no cover
+        self, index: VolumetricIndex, mode: Literal["expand", "shrink", "round"]
+    ) -> VolumetricIndex:
+        cvol = get_cv_cached(cloudpath=self.path, mip=tuple(index.resolution), **self.cv_kwargs)
+        bbox = Bbox(*tuple(zip(*((s.start, s.stop) for s in index.to_slices()))))
+        if mode == "expand":
+            bbox_aligned = bbox.expand_to_chunk_size(cvol.chunk_size, cvol.voxel_offset)
+        elif mode == "shrink":
+            bbox_aligned = bbox.shrink_to_chunk_size(cvol.chunk_size, cvol.voxel_offset)
+        elif mode == "round":
+            bbox_aligned = bbox.round_to_chunk_size(cvol.chunk_size, cvol.voxel_offset)
+        else:
+            raise NotImplementedError(
+                f"mode must be set to 'expand', 'shrink', or 'round'; received '{mode}'"
+            )
+        return VolumetricIndex(
+            resolution=index.resolution,
+            bcube=BoundingCube.from_coords(
+                IntVec3D(*bbox_aligned.minpt), IntVec3D(*bbox_aligned.maxpt), index.resolution
+            ),
+        )
+
+    def assert_idx_is_chunk_aligned(self, idx: VolumetricIndex) -> None:
+
+        # check that the idx given is chunk_aligned, and give suggestions
+        idx_expanded = self.get_chunk_aligned_index(idx, mode="expand")
+        idx_rounded = self.get_chunk_aligned_index(idx, mode="round")
+        idx_shrunk = self.get_chunk_aligned_index(idx, mode="shrink")
+
+        if idx != idx_expanded:
+            raise ValueError(
```

Comment:
Could we add a test for this? This error condition sounds actually important.

Reply:
There is already testing for this - not sure why codecov is barfing. See line 252: https://github.com/ZettaAI/zetta_utils/pull/231/files#diff-bc8abc2012f5ca7148a4aea069752161f185943809f9b30faf828d45611dbc28

Reply:
The link doesn't take me to the file. Can we confirm that that exact line is getting hit by tests by putting some `breakpoint()` or assert in there? codecov doesn't fail often

Reply:
I think it might have been catching a different `ValueError` - testing now.

Reply:
Yeah, that was it.